Multisig/Logic/Models/Chain.swift

```diff
@@ -0,0 +1,139 @@
+                       blockExplorerUrl: String,
+                       currencyName: String,
+                       currencySymbl: String,
+                       currencyDecimals: Int,
+                       transactionService: String,
+                       themeTextColor: String,
+                       themeBackgroundColor: String) {
+        dispatchPrecondition(condition: .onQueue(.main))
+        let context = App.shared.coreDataStack.viewContext
+
+        let chain = Chain(context: context)
+        chain.chainId = chainId
+        chain.chainName = chainName
+        chain.rpcUrl = rpcUrl
+        chain.blockExplorerUrl = blockExplorerUrl
+        chain.transactionService = transactionService
+        chain.theme = ChainTheme.create(textColor: themeTextColor, backgroundColor: themeBackgroundColor)
+        chain.nativeCurrency = ChainToken.create(name: currencyName, symbol: currencySymbl, decimals: currencyDecimals)
+
+        App.shared.coreDataStack.saveContext()
+    }
+
+    @discardableResult
+    static func create(_ chainInfo: SCGModels.Chain) -> Chain {
+        dispatchPrecondition(condition: .onQueue(.main))
+        let context = App.shared.coreDataStack.viewContext
+
+        let chain = Chain(context: context)
+        chain.chainId = chainInfo.chainId
+        chain.chainName = chainInfo.chainName
+        chain.rpcUrl = chainInfo.rpcUrl
+        chain.blockExplorerUrl = chainInfo.blockExplorerUrl
+        chain.transactionService = chainInfo.transactionService
+        chain.theme = ChainTheme.create(textColor: chainInfo.theme.textColor.description, backgroundColor: chainInfo.theme.backgroundColor.description)
+        chain.nativeCurrency = ChainToken.create(name: chainInfo.nativeCurrency.name, symbol: chainInfo.nativeCurrency.symbol, decimals: chainInfo.nativeCurrency.decimals)
+
+        App.shared.coreDataStack.saveContext()
+        return chain
+    }
+
+    static func updateIfExist(_ chainInfo: SCGModels.Chain) {
+        guard let chain = Chain.by(chainInfo.chainId) else { return }
+        chain.update(from: chainInfo)
+    }
+
+    static func remove(chain: Chain) {
+        let context = App.shared.coreDataStack.viewContext
+        context.delete(chain)
+        App.shared.coreDataStack.saveContext()
+    }
+
+    static func removeAll() throws {
+        for chain in all {
+            remove(chain: chain)
+        }
+    }
+}
+
+extension Chain {
+    func update(from chain: SCGModels.Chain) {
+        chainId = chainId
+        chainName = chainName
+        rpcUrl = rpcUrl
+        blockExplorerUrl = blockExplorerUrl
+        transactionService = transactionService
+        theme?.update(textColor: chain.theme.textColor.description, backgroundColor: chain.theme.backgroundColor.description)
+        nativeCurrency?.update(name: chain.nativeCurrency.name, symbol: chain.nativeCurrency.symbol, decimals: chain.nativeCurrency.decimals)
+    }
+
+    func safes() -> [Safe] {
```

Comment:
I think no need in this method. Should be added by CoreData

Reply:
I tried it but no it's not there by default